common/services/single-request.js

```diff
@@ -2,10 +2,16 @@ const dateFunctions = require('date-fns')
 const { isNil, isUndefined, mapValues, omitBy, pick } = require('lodash')
 
 const BaseService = require('./base')
-const moveService = require('./move')
+const MoveService = require('./move')
 
 const noMoveIdMessage = 'No move ID supplied'
 class SingleRequestService extends BaseService {
+  constructor(req) {
+    super()
+    this.moveService =
+      req?.services?.move || new MoveService({ apiClient: this.apiClient })
+  }
```

Comment:
A bit of service-ception going on here, we want the single request service to use the right move service. There are many ways to go about this, but they involve enhancing the constructor. I think these cases are limited, so I opted to not update the base service.

Reply:
I think this is totally acceptable. But I'd suggest 

```js
class BaseService {
  constructor(req) {
    this.req = req || {}
    this.apiClient = req.apiClient || new ApiClient()
  }
}
```

then it's just

`new MoveService(this.req)`

(sorry, it was an oversight on my part before not to have suggested stashing the request context on the instance)

Reply:
I suppose it's actually a further step towards that mythical logging we've been chatting about! :D